wcfsetup/install/files/lib/acp/page/LicensePage.class.php

```diff
@@ -0,0 +1,641 @@
+use wcf\system\database\util\PreparedStatementConditionBuilder;
+use wcf\system\io\HttpFactory;
+use wcf\system\request\LinkHandler;
+use wcf\system\WCF;
+
+/**
+ * Lists the licensed products and offers to install them.
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2023 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.0
+ */
+final class LicensePage extends AbstractPage
+{
+    public $activeMenuItem = 'wcf.acp.menu.link.package.license';
+
+    public $neededPermissions = ['admin.configuration.package.canInstallPackage'];
+
+    private array $licenseData;
+
+    private int $licenseNumber;
+
+    private array $installedPackages;
+
+    private array $installablePackages = [];
+
+    private array $packageUpdates = [];
+
+    private PackageUpdateServer $updateServer;
+
+    public function readData()
+    {
+        parent::readData();
+
+        $this->updateServer = PackageUpdateServer::getWoltLabUpdateServer();
+
+        if (!$this->hasLicenseCredentials()) {
+            return new RedirectResponse(
+                LinkHandler::getInstance()->getControllerLink(LicenseEditForm::class),
+            );
+        }
+
+        (new PackageUpdateAction([], 'refreshDatabase'))->executeAction();
+
+        $this->licenseData = $this->fetchLicenseData();
+
+        $identifiers = \array_merge(
+            \array_keys($this->licenseData['woltlab']),
+            \array_keys($this->licenseData['pluginstore'])
+        );
+        $this->installedPackages = $this->getInstalledPackages($identifiers);
+
+        $identifiers = \array_diff($identifiers, \array_keys($this->installedPackages));
+        $identifiers = $this->removeUnknownPackages($identifiers);
+
+        if ($identifiers !== []) {
+            $this->installablePackages = $this->getInstallablePackages($identifiers);
+            $this->packageUpdates = $this->getPackageUpdates($this->installablePackages);
+        }
+
+        foreach (['woltlab', 'pluginstore'] as $type) {
+            $this->licenseData[$type] = \array_filter(
+                $this->licenseData[$type],
+                function (string $package) use ($identifiers) {
+                    if (isset($this->installedPackages[$package])) {
+                        return true;
+                    }
+
+                    return \in_array($package, $identifiers);
+                },
+                \ARRAY_FILTER_USE_KEY
+            );
+
+            \uksort($this->licenseData[$type], function ($packageA, $packageB) {
+                $a = $this->installedPackages[$packageA] ?? $this->packageUpdates[$packageA];
+                $b = $this->installedPackages[$packageB] ?? $this->packageUpdates[$packageB];
+
+                if ($a->isApplication === $b->isApplication) {
+                    return $a->getName() <=> $b->getName();
+                }
+
+                return $b->isApplication <=> $a->isApplication;
```

Comment:
```suggestion
                return ($b->isApplication <=> $a->isApplication) ?: ($a->getName() <=> $b->getName());
```

Possibly with a line-break inserted before `?:`. I find this more readable, because the conditions read in the natural order priority, instead of the reverse order.